packages/server/graphql/mutations/helpers/notifications/SlackNotifier.ts

```diff
@@ -618,5 +652,68 @@ export const SlackNotifier: Notifier = {
       return handleError(res, team.id, notificationChannel)
     }
     return 'success'
+  },
+
+  async sendNotificationToUser(
+    dataLoader: DataLoaderWorker,
+    notificationId: string,
+    userId: string
+  ) {
+    const notification = await dataLoader.get('notifications').load(notificationId)
+    if (notification.type !== 'RESPONSE_MENTIONED' && notification.type !== 'RESPONSE_REPLIED') {
+      return
+    }
+
+    const meeting = await dataLoader.get('newMeetings').load(notification.meetingId)
+
+    const userSlackAuth = await getDmSlackForMeeting(dataLoader, meeting, userId)
+    if (!userSlackAuth) {
+      return
+    }
+
+    let responseId: string | undefined
+    let title: string | undefined
+    let buttonText: string | undefined
+    let body: string | undefined
+
+    if (notification.type === 'RESPONSE_REPLIED') {
+      const responses = await getTeamPromptResponsesByMeetingId(notification.meetingId)
+      responseId = responses.find(({userId: responseUserId}) => responseUserId === userId)?.id
+      const user = await dataLoader.get('users').loadNonNull(notification.authorId)
+      title = `*${user.preferredName}* replied to your response in *${meeting.name}*`
+      const comment = await dataLoader.get('comments').load(notification.commentId)
+      body = `> ${comment.plaintextContent}`
+      buttonText = 'See the discussion'
+    } else {
+      responseId = notification.responseId
+      const response = await dataLoader.get('teamPromptResponses').loadNonNull(responseId)
+      const user = await dataLoader.get('users').loadNonNull(response.userId)
+      title = `*${user.preferredName}* mentioned you in their response in *${meeting.name}*`
+      buttonText = 'See their response'
+    }
```

Comment:
+1 this could be a separate helper function that'd return `responseId`, `title`, `buttonText` and `body` for given notification

Reply:
Good call! Extracted this into a helper.